src/cdk-experimental/listbox/listbox.ts

```diff
@@ -6,26 +6,128 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+import {
+  AfterContentInit,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  Input,
+  QueryList
+} from '@angular/core';
 
+/**
+ * Directive that applies interaction patterns to an element following the aria role of option.
+ * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
+ * value is built in.
+ */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
+    '[attr.aria-selected]': '_selected',
+    '[attr.data-optionid]': '_optionId',
   }
 })
 export class CdkOption {
+  private _selected: boolean | null = null;
```

Comment:
I noticed the note about null being better for screenreaders below (I guess in order to prevent the attribute from rendering at all?), but I'm wondering if that needs to be exposed in the public interface. For instance, this could just be a boolean, and if set to false, it could render as null.

Reply:
The reason null is better is because if aria-selected is false, sometimes screen readers will read Option 1 as "Option 1, not selected" or something along those lines. It is annoying if all unselected options read like this. If aria-selected is null, then it simply reads "Option 1". Could you explain your suggestion a bit more? I don't fully understand what you're suggesting.

Reply:
I think the core question is that If we've determined that aria-selected=null is better than aria-selected="false", do we want to provide consumers with the ability to set aria-selected="false"?

The answer to this could go either way depending on the design intent of this component and existing precedent for CDK/material components.

But if we decide aria-selected=true/null is best, then we could just accept true/false here and map false to the null state.

Reply:
I see now, will change.